diff --git a/dist/index.mjs b/dist/index.mjs
index 3a1390be70464d37fc6d7fe312a22ee4c3d0e739..d1c5b2101c46262e941588a103564755eefb4662 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -587,7 +587,7 @@ async function proxyFetch(addr, input, inputInit, opts) {
 	}
 	let url;
 	let init;
-	if (input instanceof Request) {
+	if (input instanceof Request || (typeof input === 'object' && input !== null && typeof input.url === 'string')) {
 		url = new URL(input.url);
 		init = {
 			...toInit(input),
